test/val/val_cfg_test.cpp

```diff
@@ -3546,6 +3546,37 @@ OpFunctionEnd
   EXPECT_THAT(getDiagnosticString(), HasSubstr("Selection must be structured"));
 }
 
+TEST_F(ValidateCFG, LoopConditionalBranchWithoutExitBad) {
+  const std::string text = R"(
+OpCapability Shader
+OpCapability Linkage
+OpMemoryModel Logical GLSL450
+%void = OpTypeVoid
+%void_fn = OpTypeFunction %void
+%bool = OpTypeBool
+%undef = OpUndef %bool
+%func = OpFunction %void None %void_fn
+%entry = OpLabel
+OpBranch %loop
+%loop = OpLabel
+OpLoopMerge %exit %continue None
+OpBranchConditional %undef %then %else
+%then = OpLabel
+OpBranch %continue
+%else = OpLabel
+OpBranch %exit
+%continue = OpLabel
+OpBranch %loop
+%exit = OpLabel
+OpReturn
+OpFunctionEnd
```

Comment:
The Tint SPIRV-Reader currently handles this case.
When I modify this case to add an Fragment entry point, and remove Linkage, I get this:

```

$ tint --format wgsl b.spvasm
fn main_1() {
  loop {
    if (false) {
      continue;
    } else {
      break;
    }
  }
  return;
}

@fragment
fn main() {
  main_1();
}
```
Yeah, I went the extra mile to handle this extra case.  
Not passing judgement on this PR. Just FYI.